packages/playwright-core/src/server/injected/recorder.ts

```diff
@@ -494,7 +504,7 @@ export class PollingRecorder implements RecorderDelegate {
   private _pollRecorderModeTimer: NodeJS.Timeout | undefined;
 
   constructor(injectedScript: InjectedScript) {
-    this._recorder = new Recorder(injectedScript, this);
+    this._recorder = new Recorder(injectedScript, this, true);
```

Comment:
Shouldn't this take some value from externally?  How does showHighlight get set to false?

Reply:
- so the inject code is generated as part of the build, and I couldn't find a good way to pass through the param from server recorder logic to the injected logic.
- I did find that in the _pollRecorderMode logic it's communicating with the server recorder logic via  `const state = await this._embedder.__pw_recorderState().catch(() => {});` which was embedded into the browser.
- so I modified UIState to include the shouldHighlight option which is then being set in     `this._recorder.setUIState(state);`
```
 const uiState: UIState = {
        mode: this._mode,
        actionPoint,
        actionSelector,
        language: this._currentLanguage,
        showHighlight: this._showHighlight,
        testIdAttributeName: this._contextRecorder.testIdAttributeName(),
      };
```

Reply:
Coo,, thanks for the explanation. Works for me if it works.